Lib/test/test_pickle.py

```diff
@@ -699,6 +702,61 @@ def test_multiprocessing_exceptions(self):
                                  ('multiprocessing.context', name))
 
 
+class CommandLineTest(unittest.TestCase):
+    def setUp(self):
+        self.filename = tempfile.mktemp()
+        self.addCleanup(os_helper.unlink, self.filename)
+
+    @staticmethod
+    def text_normalize(string):
+        """Dedent *string* and strip it from its surrounding whitespaces.
+
+        This method is used by the other utility functions so that any
+        string to write or to match against can be freely indented.
+        """
+        return dedent(string).strip()
+
+    def set_pickle_data(self, data):
+        with open(self.filename, 'wb') as f:
+            pickle.dump(data, f)
+
+    def invoke_pickle(self, *flags):
+        output = io.StringIO()
+        with contextlib.redirect_stdout(output):
+            pickle._main(args=[*flags, self.filename])
+        return self.text_normalize(output.getvalue())
+
+    def test_invocation(self):
+        # test 'python -m pickle pickle_file'
+        data = {
+            'a': [1, 2.0, 3+4j],
+            'b': ('character string', b'byte string'),
+            'c': 'string'
+        }
+        expect = '''
+            {'a': [1, 2.0, (3+4j)],
+             'b': ('character string', b'byte string'),
+             'c': 'string'}
+        '''
+        self.set_pickle_data(data)
+
+        with self.subTest(data=data):
+            res = self.invoke_pickle()
+            expect = self.text_normalize(expect)
+            self.assertListEqual(res.splitlines(), expect.splitlines())
+
+    def test_unknown_flag(self):
+        output = io.StringIO()
+        with self.assertRaises(SystemExit):
+            # suppress argparse error message
+            with contextlib.redirect_stderr(output):
+                _ = self.invoke_pickle('--unknown')
+        msg = output.getvalue()
+        self.assertTrue(msg.startswith('usage: '),
+                        "Output does not start with 'usage: '. "
+                        f"Output was: {msg}")
```

Comment:
```suggestion
        stderr = io.StringIO()
        with self.assertRaises(SystemExit):
            # check that the parser help is shown
            with contextlib.redirect_stderr(stderr):
                _ = self.invoke_pickle('--unknown')
        self.assertStartsWith(stderr.getvalue(), 'usage: ')
```

I think we barely fit the 80 chars limit!